pkg/resourcetotelemetry/resource_to_telemetry.go

```diff
@@ -107,6 +105,16 @@ func addAttributesToExponentialHistogramDataPoints(ps pmetric.ExponentialHistogr
 }
 
 func joinAttributeMaps(from, to pcommon.Map) {
+	var capacity = from.Len() + to.Len()
+	from.Range(func(k string, v pcommon.Value) bool {
+		_, ok := to.Get(k)
+		if ok {
+			capacity--
+		}
+		return true
+	})
+
+	to.EnsureCapacity(capacity)
```

Comment:
could you benchmark just this change separately to see if it helps at all with performance? This change could also be in its own PR

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29430